nncf/common/graph/graph.py

```diff
@@ -348,20 +348,48 @@ def get_input_edges(self, node: NNCFNode) -> List[NNCFGraphEdge]:
             edges.extend(self._get_edges(from_node, node))
         return sorted(edges, key=lambda x: x.input_port_id)
 
+    def get_input_edge_by_port_id(self, node: NNCFNode, port_id: int) -> Optional[NNCFGraphEdge]:
+        """
+        Returns the input edge for a given node, where edge.input_port_id == port_id is True.
+
+        :param node: The node for which to retrieve the input edge.
+        :param port_id: The ID of the input port to filter the edges.
+        :return: An input edge connected to the specified input port ID of the
+            given node.
+        """
+        edges = [e for e in self.get_input_edges(node) if e.input_port_id == port_id]
+        if len(edges) > 1:
+            raise nncf.InternalError(
+                "Unsupported graph. More than one edge was found for a given node by the specified input port ID."
+            )
+        edge = None if len(edges) == 0 else edges[0]
```

Comment:
`None` is required here to handle case when a `node` does not have any input edges

Reply:
IMHO, this would lead to issues due to the requirement to check the return value every time.
Can we handle it somehow otherwise? For example, filtering the nodes that may miss input edge (like Constants, I think)?

Reply:
As discussed, I added an exception for cases where a node does not have input edges.